fbpic/boundaries/moving_window.py

```diff
@@ -412,3 +544,39 @@ def shift_field_array_gpu( field_array, field_buffer, n_move ):
             # Set the remaining values to zero
             if (i+n_move) >= field_array.shape[0] or (i+n_move) < 0:
                 field_buffer[i, j] = 0.
+
+    @cuda.jit('void(complex128[:,:], complex128[:], int32)')
+    def shift_spect_array_gpu( field_array, shift_factor, n_move ):
+        """
+        Shift the field 'field_array' by n_move cells on the GPU.
+        This is done in spectral space and corresponds to multiplying the
+        fields with the factor exp(i*kz_true*dz)**n_move .
+
+        Parameters
+        ----------
+        field_array: 2darray of complexs
+            Contains the value of the fields, and is modified by
+            this function
+
+        shift_factor: 1darray of complexs
+            Contains the shift array, that is multiplied to the fields in
+            spectral space to shift them by one cell in spatial space
+            ( exp(i*kz_true*dz) )
+
+        n_move: int
+            The number of cells by which the grid should be shifted
+        """
+        # Get a 2D CUDA grid
+        i, j = cuda.grid(2)
+
+        # Only access values that are actually in the array
+        if j < field_array.shape[1] and i < field_array.shape[0]:
+
+            if n_move > 0:
+                # Shift fields backward
+                field_array[i, j] = field_array[i, j] \
+                    * ( shift_factor[i] )**np.abs(n_move)
```

Comment:
No need for the `np.abs`. Then the function will be correct if `n_move` is negative.